autorest/azure/auth/auth.go

```diff
@@ -393,16 +400,30 @@ type MSIConfig struct {
 // Authorizer gets the authorizer from MSI.
 func (mc MSIConfig) Authorizer() (autorest.Authorizer, error) {
 	msiEndpoint, err := adal.GetMSIVMEndpoint()
-
 	if err != nil {
 		return nil, err
 	}
 
 	spToken, err := adal.NewServicePrincipalTokenFromMSI(msiEndpoint, mc.Resource)
-
 	if err != nil {
 		return nil, fmt.Errorf("failed to get oauth token from MSI: %v", err)
 	}
 
 	return autorest.NewBearerAuthorizer(spToken), nil
 }
+
+// IDMSConfig provides the options to get a bearer authorizer through IDMS.
```

Comment:
>IDMSConfig [](start = 3, length = 10)

IMDS should be an implementation detail for users. They just want to use MSI, they shouldnt care if its through vm extension or IMDS.

Reply:
Fixed